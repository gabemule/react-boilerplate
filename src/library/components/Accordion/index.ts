import DsAccordion from "./Accordion"
import { DsAccordionItemProps }from "@/library/types/types";

export { DsAccordion }

export type { DsAccordionItemProps }
