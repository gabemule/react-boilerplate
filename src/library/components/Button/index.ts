import DsButton from "./Button";
import { DsButtonProps }from "./Button";

export { DsButton };
export type { DsButtonProps }
