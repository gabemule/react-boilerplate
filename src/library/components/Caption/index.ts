import DsCaption from "./Caption";
import { DsCaptionProps } from "./Caption"

export { DsCaption };
export type { DsCaptionProps }
