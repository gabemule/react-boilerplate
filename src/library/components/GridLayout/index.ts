import DsGridLayout from './GridLayout'
import DsGridElement from './GridElement'
import { DsGridLayoutProps } from './GridLayout'

export { DsGridLayout, DsGridElement }
export type { DsGridLayoutProps }
