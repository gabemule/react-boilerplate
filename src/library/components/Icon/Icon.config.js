module.exports = {
  name: 'Icon',
  class: 'ds-icon',
  props: {
    color: {
      type: String,
      default: 'neutral-90',
      options: [
        '',
        'brand',
        'brand-light',
        'brand-dark',
        'brand-darkest',
        'neutral-90',
        'neutral-80',
        'neutral-60',
        'neutral-40',
        'neutral-20',
        'neutral-10',
        'neutral-05',
        'neutral-00',
        'highlight-primary',
        'highlight-primary-light',
        'highlight-primary-dark',
        'highlight-secondary',
        'highlight-secondary-light',
        'highlight-secondary-dark',
        'success',
        'success-light',
        'success-dark',
        'warning',
        'warning-light',
        'warning-dark',
        'danger',
        'danger-light',
        'danger-dark',
        'info',
        'info-light',
        'info-dark'
      ]
    },
    image: {
      type: String,
      default: '',
      options: [
        '',
        //arrows
        'arrow-back',
        'arrow-downward',
        'arrow-drop-down',
        'arrow-drop-up',
        'arrow-forward',
        'arrow-upward',
        'keyboard-arrow-down',
        'keyboard-arrow-left',
        'keyboard-arrow-right',
        'keyboard-arrow-up',
        'unfold-less',
        'unfold-more',
        //content-and-message
        'add-image',
        'article',
        'attach-file',
        'block',
        'camera',
        'chat-alert',
        'chat-bubble',
        'comment',
        'delete',
        'format-align-center',
        'format-align-justify',
        'format-align-left',
        'format-align-right',
        'format-bold',
        'format-italic',
        'format-list-bulleted',
        'format-list-numbered',
        'format-strikethrough',
        'format-underlined',
        'functions',
        'insert-code',
        'insert-emoji',
        'insert-link',
        'insert-quote',
        'insert-text',
        'keyboard-voice',
        'locked',
        'notes',
        'outline',
        'photo-add',
        'subscript',
        'superscript',
        'text-fields',
        'text-sms',
        'unlock',
        'video',
        //ecommerce
        'card',
        'code-bar',
        'discount',
        'kart-add',
        'kart-count',
        'kart-remove',
        'kart',
        'promocode',
        //essential
        'account',
        'add',
        'alarm-add',
        'alarm',
        'calendar-add',
        'calendar-check',
        'calendar',
        'cancel',
        'check-box',
        'check',
        'circle-add',
        'circle-check',
        'circle-uncheck',
        'close',
        'download',
        'edit',
        'enterprise',
        'error',
        'filter',
        'folder-add',
        'folder-remove',
        'folder',
        'info',
        'language',
        'list-settings',
        'local',
        'logout',
        'mail',
        'menu',
        'more',
        'phone',
        'refresh',
        'remove',
        'report',
        'schedule',
        'school',
        'share',
        'timer',
        'upload',
        'verified',
        'visibility-none',
        'visibility',
        'zoom-in',
        'zoom-out',
        //nav
        'activities',
        'assessment',
        'disciplines',
        'essay',
        'forum',
        'friends',
        'help',
        'home',
        'library',
        'live',
        'material',
        'mock-test',
        'notification',
        'secretary',
        'search',
        'settings',
        'shop',
        'stats',
        'study-plan',
        'terms',
        'tuttoring',
        'weekly-material',
        'work',
        //player
        'cast',
        'clock',
        'forward-10',
        'fullscreen-exit',
        'fullscreen',
        'pause',
        'picture-in-picture',
        'play',
        'playlist-add',
        'playlist',
        'replay-10',
        'replay-video',
        'replay',
        'skip-next',
        'skip-previous',
        'stop',
        'subtitles',
        'velocity',
        'volume-down',
        'volume-mute',
        'volume-off',
        'volume-up',
        //social-media
        'apple-mark',
        'apple',
        'claro-mark',
        'facebook-mark',
        'facebook',
        'google-mark',
        'google-play',
        'google',
        'instagram',
        'linkedin',
        'twitter',
        'whatsapp',
        'youtube'
      ]
    }
  }
}
