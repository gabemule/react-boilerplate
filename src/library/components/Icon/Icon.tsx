import React from "react";
// import withCustomProps from '@/library/decorators/withCustomProps'
import PropTypes from 'prop-types';

// config
import IconConfig from "./Icon.config";
// import { DsIconInterface } from "./../Icon/Icon.config";

/**
 *
 * @param {String} image
 * @param {String} color
 *
 * @return DsIcon
 */

export interface DsIconProps {
  color: string,
  image: string,
  className?: string
}

const DsIcon = ({ image, color, className }: DsIconProps) => {
  const iconClass = IconConfig.props.image.options.find((value: string) => value === image) && image;
  const iconColor = color && IconConfig.props.color.options!.find((value: string) => value === color) && `${IconConfig.class}__color--${color}`;

  return (
    <div
      data-testid="text-icon"
      className={
        [
          IconConfig.class,
          `${IconConfig.class}--${iconClass}`,
          iconColor,
          className && className
        ].filter(classes => classes).join(" ")}
    />
  );
};

DsIcon.propTypes = {
  color: PropTypes.oneOf(IconConfig.props.color.options),
  image: PropTypes.oneOf(IconConfig.props.image.options),
}

DsIcon.defaultProps = {
  color: IconConfig.props.color.default,
  image: IconConfig.props.image.default,
}

export default DsIcon;
