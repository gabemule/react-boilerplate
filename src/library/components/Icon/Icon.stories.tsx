import React from "react";
import { Meta, Story } from "@storybook/react"

// config
import IconConfig from "./Icon.config";

// components
import { DsIcon, DsIconProps } from "@/library/components/Icon";
import { ThemeProvider, ThemeProviderProps } from "@/library/components/ThemeProvider";

const typesOfIcons = [
  { title: "Essential", start: 57, qtd: 100 },
  { title: "Player", start: 124, qtd: 145 },
  { title: "Nav", start: 101, qtd: 123 },
  { title: "Content and Message", start: 13, qtd: 48 },
  { title: "Social Media", start: 146, qtd: 158 },
  { title: "Arrows", start: 1, qtd: 12 },
  { title: "Ecommerce", start: 49, qtd: 56 },
];

const ComponentProps: Meta = {
  title: "Interação Básica/Icon/Icon",
  component: DsIcon,
  argTypes: {
    image: {
      description: "Nome do Icon desejado, deve seguir o padrão 'Nome-do-Icon'. Ex.: image='arrow-back' ",
      table: {
        category: "Image props",
        type: { summary: "string" },
        defaultValue: IconConfig.props.image.default,
      },
      options: IconConfig.props.image.options,
      control: { type: "select" }
    },
    color: {
      description: "Define a cor do icon, o padrão é preto",
      table: {
        category: "Style props",
        type: { summary: "string" },
        defaultValue: { summary: IconConfig.props.color.default },
      },
      defaultValue: IconConfig.props.color.default,
      options: IconConfig.props.color.options,
      control: { type: "select" }
    }
  },
};

export default ComponentProps;

const Template: Story<ThemeProviderProps & DsIconProps> = (args) => (
  <ThemeProvider library={"core"} theme={"light"} >
    <DsIcon {...args} />
  </ThemeProvider>
);

export const Default = Template.bind({});

Default.args = {
  image: IconConfig.props.image.default,
  color: IconConfig.props.color.default,
};

const ListTemplate: Story<ThemeProviderProps & DsIconProps> = (args) => {
  return (
    <ThemeProvider library={"core"} theme={"light"}>
      <div style={{ display: "flex", flexWrap: "wrap", borderWidth: "1px", borderColor: "black", borderStyle: "solid" }}>
        {typesOfIcons.map((type, index) =>
          <>
            <h3 style={{ margin: '10px', fontFamily: 'sans-serif' }}>{type.title}</h3>
            <div key={index} style={{ display: "flex", flexWrap: "wrap", borderWidth: "1px", borderColor: "black", borderStyle: "solid", margin: 10, width: "100%" }}>
              {IconConfig.props.image.options
                .filter((item: String, index: Number) => index >= type.start && index <= type.qtd)
                .map((item: string) =>
                  <div key={item} style={{ margin: 10, display: "flex", flexDirection: "column", alignItems: "center" }}>
                    <DsIcon color={args.color} image={item} />
                    <p style={{ fontSize: "0.7rem" }}>{item}</p>
                  </div>
                )}
            </div>
          </>
        )}
      </div>
    </ThemeProvider>
  );
};



export const List = ListTemplate.bind({});
List.args = {}
