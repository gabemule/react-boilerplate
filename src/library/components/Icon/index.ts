import DsIcon from "./Icon";
import { DsIconProps } from "./Icon"

export { DsIcon };
export type { DsIconProps };
