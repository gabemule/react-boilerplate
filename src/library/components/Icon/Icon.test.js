import React from 'react'
import { shallow, render } from 'enzyme'

// components
import { DsIcon } from '@/library/components/Icon'

// config
import IconConfig from './Icon.config'

// constants
const IconClass = IconConfig.class
const IconPropImage = IconConfig.props.image.options[1]
const IconPropColor = IconConfig.props.color.default

describe('Icon Component', () => {
  it('Matches Snapshot', () => {
    const wrapper = render(<DsIcon image={IconPropImage} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    const wrapper = shallow(<DsIcon image={IconPropImage} />)
    expect(wrapper).toBeTruthy()
    expect(wrapper.find(IconClass)).toBeTruthy()
  })

  it('Loads the Component HTML with primary color', () => {
    const wrapper = shallow(
      <DsIcon color={IconPropColor} image={IconPropImage} />
    )
    expect(wrapper.find(`${IconClass}__${IconPropColor}`)).toBeTruthy()
  })
})
