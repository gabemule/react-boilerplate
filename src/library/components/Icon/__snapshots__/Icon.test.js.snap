// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component Matches Snapshot 1`] = `
LoadedCheerio {
  "0": <div
    class="ds-icon ds-icon--arrow-back ds-icon__color--neutral-90"
    data-testid="text-icon"
  />,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
